src/models/admin_permissions.model.ts

```diff
@@ -0,0 +1,11 @@
+export interface Permissions {
+  isMentor?: boolean;
+  isOrgManager?: boolean;
+  isCentreManager?: boolean;
+  isAdmin?: boolean;
+}
+
+export default interface AdminPermission extends Permissions {
+  id?: string;
+  userId?: string;
+}
```

Comment:
approvedUser object contains those permission fields (isOrgManager, isCentreManger etc.), the function requires and uses only these permission fields from the passed parameter.

Reply:
So, lets say if I had called `permissions.password` inside the `addPermission` function definition, would typescript have disallowed it since `password` is not defined for `Permissions`

Reply:
Correct. Typescript will throw an error in this case.